client.go

```diff
@@ -178,6 +178,9 @@ func (d processInOption) Value() interface{} { return time.Duration(d) }
 // ErrDuplicateTask error only applies to tasks enqueued with a Unique option.
 var ErrDuplicateTask = errors.New("task already exists")
 
+// ErrEmptyTypeTask indicates that task's typename is not specified.
+var ErrEmptyTypeTask = errors.New("task typename not specified")
```

Comment:
I don't expect many users encountering this error, so surfacing a sentinel error value via public API seems like an overkill. Let's not add this to package API yet. We can always add this to public API if needed :)

Also, another nit-pick: can we change the error message to `"task typename cannot be empty"` (reason: Technically user has *specified* typename, it's just empty)

Reply:
Thanks for the feedback. I fully agree that the error message you suggested sounds better. Adjusted it as requested.